src/command/render/output.ts

```diff
@@ -158,10 +160,20 @@ export function outputRecipe(
           const outputMd = partitionYamlFrontMatter(
             Deno.readTextFileSync(outputFile),
           );
+          // remove _quarto metadata
+          //
+          // this is required to avoid tests breaking due to the
+          // _quarto regexp tests finding themselves in the output
+          const yaml = parseYaml(
+            inputMd.yaml.replace(/^---+\n/m, "").replace(/\n---+\n*$/m, "\n"),
+          ) as Record<string, unknown>;
+          delete yaml._quarto;
+          const yamlString = `---\n${stringifyYaml(yaml)}---\n`;
```

Comment:
This just reminded me of another place where we are supposed to do this I believe. 
https://github.com/quarto-dev/quarto-cli/blob/6821f7197aff2b18132e3d1de8720ddaf7599ec9/src/command/render/pandoc.ts#L1087-L1091

Would it mean the above does not work anymore? Or was not needed? 

only asking to clean our code base if we can. 

And I am surprised we got hit by this `ensureFileRegexMatches()` problem only now... 🤯  Good find!

Reply:
No, those are _separate_ code paths. We need that to be deleted because sometimes the metadata is emitted by Pandoc. But sometimes it's not: see what happens with a plain `markdown` output. It's really annoying, and we should fix in the future.

Reply:
Specifically, the problem is in `hugo-md` and `docusaurus-md`, where the metadata block preservation happens through that code in `output.ts`. It's also annoying (and almost certainly a separate bug) that the frontmatter block that is preserved, in that situation, is only the very first one (like RMarkdown).

But we're not going to pull on that thread right now.